# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/cocoa/nsMacFinderProgress.mm
# Commit: 40fe811eab33
# Full Hash: 40fe811eab33812e9536ceaaeeedac0c8be18743
# Author: Christoph Walcher <christoph-wa@gmx.de>
# Date: 2020-03-30 22:01:01
# Description:
#   Bug 1624251 - Fix use-after-free in nsMacFinderProgress r=spohl
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D68667
# ==============================================================================

diff -r 2a1b5f29d43d -r 40fe811eab33 widget/cocoa/nsMacFinderProgress.mm
--- a/widget/cocoa/nsMacFinderProgress.mm	Mon Mar 30 13:40:22 2020 +0000
+++ b/widget/cocoa/nsMacFinderProgress.mm	Sat Mar 28 01:46:29 2020 +0000
@@ -17,7 +17,7 @@
 
 nsMacFinderProgress::~nsMacFinderProgress() {
   if (mProgress) {
-    [mProgress.cancellationHandler release];
+    [mProgress unpublish];
     [mProgress release];
   }
 }
@@ -78,8 +78,6 @@
 
   if (mProgress) {
     [mProgress unpublish];
-    [mProgress release];
-    mProgress = nil;
   }
 
   return NS_OK;
